### PR TITLE
Create game with Validations

### PR DESCRIPTION
**Pull request by ObeydKhan**
Added Formik, Yup and Material UI. Game creation working end to end with proper validations.

### CODE REVIEW

File: src/app/components/dashboard/FormikField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { ErrorMessage, Field } from "formik";
+// @ts-ignore
```
Comment (alanjb) [inline src/app/components/dashboard/FormikField/index.tsx]: We shouldn't have ts-ignore here. Instead, we should update the tsconfig.json file.

File: src/game/components/CreateGameModal.tsx [hunk 2]
```diff
@@ -1,60 +1,80 @@
 import axios from 'axios';
-import React, { Component } from 'react';
-import {Button, Form, FormGroup, Input, Label, Modal, ModalBody, ModalFooter, ModalHeader} from 'reactstrap';
+import React from 'react';
+import { Modal, ModalBody, ModalHeader} from 'reactstrap';
+import { Formik, Form} from 'formik';
+import FormikField from "../../app/components/dashboard/FormikField";
+//@ts-ignore
```
Comment (alanjb) [inline src/game/components/CreateGameModal.tsx]: same as above

File: src/game/components/CreateGameModal.tsx [hunk 3]
```diff
@@ -1,60 +1,80 @@
...
+import Button from '@material-ui/core/Button';
+
```
Comment (alanjb) [inline src/game/components/CreateGameModal.tsx:7]: extra space

**Review by alanjb — COMMENTED** [review]
Looks good, just a few comments

**Comment by ObeydKhan** [comment]
> Looks good, just a few comments

For some weird reason, don't see the ts errors after removing @ts-ignore. Builds and runs fine.

**Review by alanjb — APPROVED** [review]

**Pull request merged**